specification/machinelearning/data-plane/Azure.AI.ChatProtocol/preview/2023-10-01-preview/generated.json

```
@@ -0,0 +1,325 @@
+                      },
+                      "session_state": {
+                        "description": "Backend-specific information for the tracking of a session.",
+                        "x-ms-client-name": "sessionState"
+                      }
+                    },
+                    "required": [
+                      "content",
+                      "role"
+                    ]
+                  },
+                  "x-ms-identifiers": []
+                },
+                "stream": {
+                  "type": "boolean",
+                  "description": "Indicates whether the completion is a streaming or non-streaming completion.",
+                  "enum": [
+                    true
+                  ]
+                },
+                "session_state": {
+                  "description": "Backend-specific information for the tracking of a session.",
+                  "x-ms-client-name": "sessionState"
+                },
+                "context": {
+                  "type": "object",
+                  "description": "Backend-specific context or arguments.",
+                  "additionalProperties": {}
+                }
+              },
+              "required": [
+                "messages",
+                "stream"
+              ]
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request has succeeded.",
+            "schema": {
+              "type": "object",
+              "description": "A single response to a streaming completion request.",
+              "properties": {
+                "choices": {
+                  "type": "array",
+                  "description": "The collection of choice deltas received in this chunk.",
+                  "items": {
+                    "type": "object",
+                    "description": "The representation of an incremental choice received in a streaming completion.",
+                    "properties": {
+                      "index": {
+                        "type": "integer",
+                        "format": "int64",
+                        "description": "The index of the of the chat choice, relative to the other choices in the same completion."
+                      },
+                      "delta": {
+                        "type": "object",
+                        "description": "The partial message received for this choice.",
+                        "properties": {
+                          "content": {
+                            "type": "string",
+                            "description": "An incremental part of the text associated with the message."
+                          },
+                          "role": {
+                            "$ref": "#/definitions/ChatRole",
+                            "description": "The role associated with the message."
+                          },
+                          "session_state": {
```

Azure convention is that property names are lower-camel-case.
```suggestion
                          "sessionState": {
```

---

Yes, I'm aware about the camel-case convention. The problem is that for ignite we have to deliver a library that can talk with a service that is already in place. After Ignite I plan to work with the team to move this towards using camel here (amongst other changes)